In [3]:
import pandas as pd
import numpy as np
import matplotlib.pyplot as plt
import seaborn as sns
%matplotlib inline

##Install BeautifulSoup and Requests

In [6]:
!pip install BeautifulSoup
!pip install Requests

  Using cached https://files.pythonhosted.org/packages/1e/ee/295988deca1a5a7accd783d0dfe14524867e31abb05b6c0eeceee49c759d/BeautifulSoup-3.2.1.tar.gz
    ERROR: Complete output from command python setup.py egg_info:
    ERROR: Traceback (most recent call last):
      File "<string>", line 1, in <module>
      File "/tmp/pip-install-5619npzw/BeautifulSoup/setup.py", line 22
        print "Unit tests have failed!"
                                      ^
    SyntaxError: Missing parentheses in call to 'print'. Did you mean print("Unit tests have failed!")?
    ----------------------------------------
ERROR: Command "python setup.py egg_info" failed with error code 1 in /tmp/pip-install-5619npzw/BeautifulSoup/


In [7]:
!pip install BeautifulSoup4

In [8]:
pip install lxml

Note: you may need to restart the kernel to use updated packages.


## Import Beautiful Soup

In [9]:
from urllib.request import urlopen
import requests
from bs4 import BeautifulSoup

print('Done')

Done


In [10]:
url = requests.get("https://en.wikipedia.org/wiki/List_of_postal_codes_of_Canada:_M").text

### Map URL and interpret HTML Table

In [11]:
soup = BeautifulSoup(url,'lxml')
type(soup)
#print(soup.prettify())

bs4.BeautifulSoup

## Identify Postal Code Table

In [13]:
My_table = soup.find('table',{'class':'wikitable sortable'})
#My_table

In [14]:
import csv

## Fetch Content from HTML Table into Rows and Columns

In [15]:
list_of_rows= []   
rows  = My_table.find_all('tr')

for row in rows:
    list_of_cells = []
    for cell in row.find_all('td'):
        list_of_cells.append(cell.text)
    list_of_rows.append(list_of_cells)
list_of_rows

[[],
 ['M1A', 'Not assigned', 'Not assigned\n'],
 ['M2A', 'Not assigned', 'Not assigned\n'],
 ['M3A', 'North York', 'Parkwoods\n'],
 ['M4A', 'North York', 'Victoria Village\n'],
 ['M5A', 'Downtown Toronto', 'Harbourfront\n'],
 ['M5A', 'Downtown Toronto', 'Regent Park\n'],
 ['M6A', 'North York', 'Lawrence Heights\n'],
 ['M6A', 'North York', 'Lawrence Manor\n'],
 ['M7A', "Queen's Park", 'Not assigned\n'],
 ['M8A', 'Not assigned', 'Not assigned\n'],
 ['M9A', 'Etobicoke', 'Islington Avenue\n'],
 ['M1B', 'Scarborough', 'Rouge\n'],
 ['M1B', 'Scarborough', 'Malvern\n'],
 ['M2B', 'Not assigned', 'Not assigned\n'],
 ['M3B', 'North York', 'Don Mills North\n'],
 ['M4B', 'East York', 'Woodbine Gardens\n'],
 ['M4B', 'East York', 'Parkview Hill\n'],
 ['M5B', 'Downtown Toronto', 'Ryerson\n'],
 ['M5B', 'Downtown Toronto', 'Garden District\n'],
 ['M6B', 'North York', 'Glencairn\n'],
 ['M7B', 'Not assigned', 'Not assigned\n'],
 ['M8B', 'Not assigned', 'Not assigned\n'],
 ['M9B', 'Etobicoke', 'Cloverdale

In [17]:


outfile=open("./postcodes.csv","w")
writer =csv.writer(outfile)
writer.writerow(["Postal Code","Borough","Neighbourhood"])
writer.writerows(list_of_rows)

### Generate CSV file and migrate the HTML List into CSV file

In [18]:
#import necessary modules

with open('./postcodes.csv','rt')as f:
  data = csv.reader(f)
  for row in data:
        print(row)

['Postal Code', 'Borough', 'Neighbourhood']
[]
['M1A', 'Not assigned', 'Not assigned\n']
['M2A', 'Not assigned', 'Not assigned\n']
['M3A', 'North York', 'Parkwoods\n']
['M4A', 'North York', 'Victoria Village\n']
['M5A', 'Downtown Toronto', 'Harbourfront\n']
['M5A', 'Downtown Toronto', 'Regent Park\n']
['M6A', 'North York', 'Lawrence Heights\n']
['M6A', 'North York', 'Lawrence Manor\n']
['M7A', "Queen's Park", 'Not assigned\n']
['M8A', 'Not assigned', 'Not assigned\n']
['M9A', 'Etobicoke', 'Islington Avenue\n']
['M1B', 'Scarborough', 'Rouge\n']
['M1B', 'Scarborough', 'Malvern\n']
['M2B', 'Not assigned', 'Not assigned\n']
['M3B', 'North York', 'Don Mills North\n']
['M4B', 'East York', 'Woodbine Gardens\n']
['M4B', 'East York', 'Parkview Hill\n']
['M5B', 'Downtown Toronto', 'Ryerson\n']
['M5B', 'Downtown Toronto', 'Garden District\n']
['M6B', 'North York', 'Glencairn\n']
['M7B', 'Not assigned', 'Not assigned\n']
['M8B', 'Not assigned', 'Not assigned\n']
['M9B', 'Etobicoke', 'Cloverdale\n'

### Read CSV File into Data frame

In [20]:
import pandas as pd
result=pd.read_csv('./postcodes.csv')
df = pd.DataFrame(result)
df

Postal Code           Borough            Neighbourhood
0           M1A      Not assigned           Not assigned\n
1           M2A      Not assigned           Not assigned\n
2           M3A        North York              Parkwoods\n
3           M4A        North York       Victoria Village\n
4           M5A  Downtown Toronto           Harbourfront\n
..          ...               ...                      ...
283         M8Z         Etobicoke              Mimico NW\n
284         M8Z         Etobicoke     The Queensway West\n
285         M8Z         Etobicoke  Royal York South West\n
286         M8Z         Etobicoke         South of Bloor\n
287         M9Z      Not assigned           Not assigned\n

[288 rows x 3 columns]

## Clean the table by removing EoL

In [21]:
df['Neighbourhood'].replace(r'\s+|\\n', ' ', regex=True, inplace=True)
print(df)

    Postal Code           Borough           Neighbourhood
0           M1A      Not assigned           Not assigned 
1           M2A      Not assigned           Not assigned 
2           M3A        North York              Parkwoods 
3           M4A        North York       Victoria Village 
4           M5A  Downtown Toronto           Harbourfront 
..          ...               ...                     ...
283         M8Z         Etobicoke              Mimico NW 
284         M8Z         Etobicoke     The Queensway West 
285         M8Z         Etobicoke  Royal York South West 
286         M8Z         Etobicoke         South of Bloor 
287         M9Z      Not assigned           Not assigned 

[288 rows x 3 columns]


# Remove Boroughs from data frame where value is Not assigned

In [22]:
df1= df[df['Borough']!= 'Not assigned']
df1

Postal Code           Borough              Neighbourhood
2           M3A        North York                 Parkwoods 
3           M4A        North York          Victoria Village 
4           M5A  Downtown Toronto              Harbourfront 
5           M5A  Downtown Toronto               Regent Park 
6           M6A        North York          Lawrence Heights 
..          ...               ...                        ...
282         M8Z         Etobicoke  Kingsway Park South West 
283         M8Z         Etobicoke                 Mimico NW 
284         M8Z         Etobicoke        The Queensway West 
285         M8Z         Etobicoke     Royal York South West 
286         M8Z         Etobicoke            South of Bloor 

[211 rows x 3 columns]

## 77 rows were not considered where Borough = Not Assigned

In [23]:
df1.sort_values(by=['Postal Code'])

Postal Code      Borough    Neighbourhood
11          M1B  Scarborough           Rouge 
12          M1B  Scarborough         Malvern 
29          M1C  Scarborough      Port Union 
28          M1C  Scarborough      Rouge Hill 
27          M1C  Scarborough  Highland Creek 
..          ...          ...              ...
232         M9V    Etobicoke     Mount Olive 
234         M9V    Etobicoke   South Steeles 
235         M9V    Etobicoke     Thistletown 
233         M9V    Etobicoke     Silverstone 
245         M9W    Etobicoke       Northwest 

[211 rows x 3 columns]

In [43]:
## Importing data from CSV file for all postal codes
import pandas as pd
coord = pd.read_csv('http://cocl.us/Geospatial_data')
coord

Postal Code   Latitude  Longitude
0           M1B  43.806686 -79.194353
1           M1C  43.784535 -79.160497
2           M1E  43.763573 -79.188711
3           M1G  43.770992 -79.216917
4           M1H  43.773136 -79.239476
..          ...        ...        ...
98          M9N  43.706876 -79.518188
99          M9P  43.696319 -79.532242
100         M9R  43.688905 -79.554724
101         M9V  43.739416 -79.588437
102         M9W  43.706748 -79.594054

[103 rows x 3 columns]

In [63]:
#Generating DataSet with Lat Long

result3 = pd.merge(df1,coord,on='Postal Code',how='left')
result3

Postal Code           Borough  \
0           M3A        North York   
1           M4A        North York   
2           M5A  Downtown Toronto   
3           M5A  Downtown Toronto   
4           M6A        North York   
5           M6A        North York   
6           M7A      Queen's Park   
7           M9A         Etobicoke   
8           M1B       Scarborough   
9           M1B       Scarborough   
10          M3B        North York   
11          M4B         East York   
12          M4B         East York   
13          M5B  Downtown Toronto   
14          M5B  Downtown Toronto   
15          M6B        North York   
16          M9B         Etobicoke   
17          M9B         Etobicoke   
18          M9B         Etobicoke   
19          M9B         Etobicoke   
20          M9B         Etobicoke   
21          M1C       Scarborough   
22          M1C       Scarborough   
23          M1C       Scarborough   
24          M3C        North York   
25          M3C        North York   
26          M4C         East York   
27          M5C  Downtown Toronto   
28          M6C              York   
29          M9C         Etobicoke   
30          M9C         Etobicoke   
31          M9C         Etobicoke   
32          M9C         Etobicoke   
33          M1E       Scarborough   
34          M1E       Scarborough   
35          M1E       Scarborough   
36          M4E      East Toronto   
37          M5E  Downtown Toronto   
38          M6E              York   
39          M1G       Scarborough   
40          M4G         East York   
41          M5G  Downtown Toronto   
42          M6G  Downtown Toronto   
43          M1H       Scarborough   
44          M2H        North York   
45          M3H        North York   
46          M3H        North York   
47          M3H        North York   
48          M4H         East York   
49          M5H  Downtown Toronto   
50          M5H  Downtown Toronto   
51          M5H  Downtown Toronto   
52          M6H      West Toronto   
53          M6H      West Toronto   
54          M1J       Scarborough   
55          M2J        North York   
56          M2J        North York   
57          M2J        North York   
58          M3J        North York   
59          M3J        North York   
60          M4J         East York   
61          M5J  Downtown Toronto   
62          M5J  Downtown Toronto   
63          M5J  Downtown Toronto   
64          M6J      West Toronto   
65          M6J      West Toronto   
66          M1K       Scarborough   
67          M1K       Scarborough   
68          M1K       Scarborough   
69          M2K        North York   
70          M3K        North York   
71          M3K        North York   
72          M4K      East Toronto   
73          M4K      East Toronto   
74          M5K  Downtown Toronto   
75          M5K  Downtown Toronto   
76          M6K      West Toronto   
77          M6K      West Toronto   
78          M6K      West Toronto   
79          M1L       Scarborough   
80          M1L       Scarborough   
81          M1L       Scarborough   
82          M2L        North York   
83          M2L        North York   
84          M3L        North York   
85          M4L      East Toronto   
86          M4L      East Toronto   
87          M5L  Downtown Toronto   
88          M5L  Downtown Toronto   
89          M6L        North York   
90          M6L        North York   
91          M6L        North York   
92          M9L        North York   
93          M1M       Scarborough   
94          M1M       Scarborough   
95          M1M       Scarborough   
96          M2M        North York   
97          M2M        North York   
98          M3M        North York   
99          M4M      East Toronto   
100         M5M        North York   
101         M5M        North York   
102         M6M              York   
103         M6M              York   
104         M6M              York   
105         M6M              York   
106         M9M        North York   
107     

## CHeck rows where Neighbourhood values is Not Assigned

In [45]:
print("Check Not Assigned values in Neighbourhood column:\n") 
df2 = result3['Neighbourhood'].str.contains('Not assigned') 
df2

Check Not Assigned values in Neighbourhood column:



0      False
1      False
2      False
3      False
4      False
       ...  
206    False
207    False
208    False
209    False
210    False
Name: Neighbourhood, Length: 211, dtype: bool

###Results shows there are no rows where Neighbourhood value is not assigned 

## Now merge neighbourhood with same Post codes and Borough names into single row

In [46]:
df2 = (result3.groupby(['Postal Code', 'Borough','Latitude','Longitude'])['Neighbourhood'].apply(list))
df2

Postal Code  Borough      Latitude   Longitude 
M1B          Scarborough  43.806686  -79.194353                                   [Rouge , Malvern ]
M1C          Scarborough  43.784535  -79.160497          [Highland Creek , Rouge Hill , Port Union ]
M1E          Scarborough  43.763573  -79.188711               [Guildwood , Morningside , West Hill ]
M1G          Scarborough  43.770992  -79.216917                                            [Woburn ]
M1H          Scarborough  43.773136  -79.239476                                         [Cedarbrae ]
                                                                         ...                        
M9N          York         43.706876  -79.518188                                            [Weston ]
M9P          Etobicoke    43.696319  -79.532242                                         [Westmount ]
M9R          Etobicoke    43.688905  -79.554724    [Kingsview Village , Martin Grove Gardens , Ri...
M9V          Etobicoke    43.739416  -79.58

## find shape of the cleaned data set

In [47]:
df2.shape

(103,)

## On Clubbing the Neighbourhoods, total data set size is 103 rows

In [48]:
!pip install geopy

In [49]:
import numpy as np # library to handle data in a vectorized manner

import pandas as pd # library for data analsysis
pd.set_option('display.max_columns', None)
pd.set_option('display.max_rows', None)

import json # library to handle JSON files

#!conda install -c conda-forge geopy --yes # uncomment this line if you haven't completed the Foursquare API lab
from geopy.geocoders import Nominatim # convert an address into latitude and longitude values

import requests # library to handle requests
from pandas.io.json import json_normalize # tranform JSON file into a pandas dataframe

# Matplotlib and associated plotting modules
import matplotlib.cm as cm
import matplotlib.colors as colors

# import k-means from clustering stage
from sklearn.cluster import KMeans

#!conda install -c conda-forge folium=0.5.0 --yes # uncomment this line if you haven't completed the Foursquare API lab
import folium # map rendering library

print('Libraries imported.')

Libraries imported.


In [51]:
print('The dataframe has {} boroughs and {} neighborhoods.'.format(
        len(result3['Borough'].unique()),
        result3.shape[0]
    )
)

The dataframe has 11 boroughs and 211 neighborhoods.


In [55]:
address = 'Toronto,Canada'

geolocator = Nominatim(user_agent="toronto_explorer")
location = geolocator.geocode(address)
latitude = location.latitude
longitude = location.longitude
print('The geograpical coordinate of Toronto are {}, {}.'.format(latitude, longitude))

The geograpical coordinate of Toronto are 43.653963, -79.387207.


In [57]:
# create map of New York using latitude and longitude values
map_Toronto = folium.Map(location=[latitude, longitude], zoom_start=10)   
map_Toronto

In [66]:
# add markers to map
for lat, lng, borough, neighbourhood in zip(result3['Latitude'], result3['Longitude'], result3['Borough'], result3['Neighbourhood']):
    
    label = '{}, {}'.format(borough,neighbourhood)
    label = folium.Popup(label, parse_html=True)
    folium.CircleMarker(
        [lat, lng],
        radius=5,
        popup=label,
        color='blue',
        fill=True,
        fill_color='#3186cc',
        fill_opacity=0.7,
        parse_html=False).add_to(map_Toronto)  
    
map_Toronto

In [70]:
toronto_data = result3[result3['Borough'].str.contains('Toronto')].reset_index(drop=True)
toronto_data

Postal Code           Borough  \
0          M5A  Downtown Toronto   
1          M5A  Downtown Toronto   
2          M5B  Downtown Toronto   
3          M5B  Downtown Toronto   
4          M5C  Downtown Toronto   
5          M4E      East Toronto   
6          M5E  Downtown Toronto   
7          M5G  Downtown Toronto   
8          M6G  Downtown Toronto   
9          M5H  Downtown Toronto   
10         M5H  Downtown Toronto   
11         M5H  Downtown Toronto   
12         M6H      West Toronto   
13         M6H      West Toronto   
14         M5J  Downtown Toronto   
15         M5J  Downtown Toronto   
16         M5J  Downtown Toronto   
17         M6J      West Toronto   
18         M6J      West Toronto   
19         M4K      East Toronto   
20         M4K      East Toronto   
21         M5K  Downtown Toronto   
22         M5K  Downtown Toronto   
23         M6K      West Toronto   
24         M6K      West Toronto   
25         M6K      West Toronto   
26         M4L      East Toronto   
27         M4L      East Toronto   
28         M5L  Downtown Toronto   
29         M5L  Downtown Toronto   
30         M4M      East Toronto   
31         M4N   Central Toronto   
32         M5N   Central Toronto   
33         M4P   Central Toronto   
34         M5P   Central Toronto   
35         M5P   Central Toronto   
36         M6P      West Toronto   
37         M6P      West Toronto   
38         M4R   Central Toronto   
39         M5R   Central Toronto   
40         M5R   Central Toronto   
41         M5R   Central Toronto   
42         M6R      West Toronto   
43         M6R      West Toronto   
44         M4S   Central Toronto   
45         M5S  Downtown Toronto   
46         M5S  Downtown Toronto   
47         M6S      West Toronto   
48         M6S      West Toronto   
49         M4T   Central Toronto   
50         M4T   Central Toronto   
51         M5T  Downtown Toronto   
52         M5T  Downtown Toronto   
53         M5T  Downtown Toronto   
54         M4V   Central Toronto   
55         M4V   Central Toronto   
56         M4V   Central Toronto   
57         M4V   Central Toronto   
58         M4V   Central Toronto   
59         M5V  Downtown Toronto   
60         M5V  Downtown Toronto   
61         M5V  Downtown Toronto   
62         M5V  Downtown Toronto   
63         M5V  Downtown Toronto   
64         M5V  Downtown Toronto   
65         M5V  Downtown Toronto   
66         M4W  Downtown Toronto   
67         M5W  Downtown Toronto   
68         M4X  Downtown Toronto   
69         M4X  Downtown Toronto   
70         M5X  Downtown Toronto   
71         M5X  Downtown Toronto   
72         M4Y  Downtown Toronto   
73         M7Y      East Toronto   

                                        Neighbourhood   Latitude  Longitude  
0                                       Harbourfront   43.654260 -79.360636  
1                                        Regent Park   43.654260 -79.360636  
2                                            Ryerson   43.657162 -79.378937  
3                                    Garden District   43.657162 -79.378937  
4                                     St. James Town   43.651494 -79.375418  
5                                        The Beaches   43.676357 -79.293031  
6                                        Berczy Park   43.644771 -79.373306  
7                                 Central Bay Street   43.657952 -79.387383  
8                                           Christie   43.669542 -79.422564  
9                                           Adelaide   43.650571 -79.384568  
10                                              King   43.650571 -79.384568  
11                                          Richmond   43.650571 -79.384568  
12                                Dovercourt Village   43.669005 -79.442259  
13                                          Dufferin   43.669005 -79.442259  
14                                 Harbourfront East   43.640816 -79.381752  
15                                   Toronto Islands  

In [71]:
address = 'Toronto,To'

geolocator = Nominatim(user_agent="To_explorer")
location = geolocator.geocode(address)
latitude = location.latitude
longitude = location.longitude
print('The geograpical coordinate of Manhattan are {}, {}.'.format(latitude, longitude))

The geograpical coordinate of Manhattan are 43.6523873, -79.3835641.


In [86]:
map_TorontoCity = folium.Map(location=[latitude, longitude], zoom_start=11)

# add markers to map
for lat, lng, label in zip(toronto_data['Latitude'], toronto_data['Longitude'], toronto_data['Neighbourhood']):
    label = folium.Popup(label, parse_html=True)
    folium.CircleMarker(
        [lat, lng],
        radius=5,
        popup=label,
        color='blue',
        fill=True,
        fill_color='#3186cc',
        fill_opacity=0.7,
        parse_html=False,
    clustered_marker = True).add_to(map_TorontoCity)  
    
map_TorontoCity

In [74]:
CLIENT_ID = '0PFZU1J2VP3OZMYAZC24EA2MNEPLN4XQHYBMJOB3EUY4GK2H' # your Foursquare ID
CLIENT_SECRET = 'FGTNJ055D2UTPZJIPYBPNPLWSNDSIJJU3WA4AENELLDX52ID' # your Foursquare Secret
VERSION = '20180605' # Foursquare API version

print('Your credentails:')
print('CLIENT_ID: ' + CLIENT_ID)
print('CLIENT_SECRET:' + CLIENT_SECRET)

Your credentails:
CLIENT_ID: 0PFZU1J2VP3OZMYAZC24EA2MNEPLN4XQHYBMJOB3EUY4GK2H
CLIENT_SECRET:FGTNJ055D2UTPZJIPYBPNPLWSNDSIJJU3WA4AENELLDX52ID


In [76]:
toronto_data.loc[0, 'Neighbourhood']

'Harbourfront '

In [78]:
neighbourhood_latitude = toronto_data.loc[0, 'Latitude'] # neighborhood latitude value
neighbourhood_longitude = toronto_data.loc[0, 'Longitude'] # neighborhood longitude value

neighbourhood_name = toronto_data.loc[0, 'Neighbourhood'] # neighborhood name

print('Latitude and longitude values of {} are {}, {}.'.format(neighbourhood_name, 
                                                               neighbourhood_latitude, 
                                                               neighbourhood_longitude))

Latitude and longitude values of Harbourfront  are 43.6542599, -79.3606359.


In [79]:
LIMIT=100
RADIUS=500
url = 'https://api.foursquare.com/v2/venues/explore?&client_id={}&client_secret={}&v={}&ll={},{}&radius={}&limit={}'.format(
    CLIENT_ID, 
    CLIENT_SECRET, 
    VERSION, 
    neighbourhood_latitude, 
    neighbourhood_longitude, 
    RADIUS, 
    LIMIT)

url

'https://api.foursquare.com/v2/venues/explore?&client_id=0PFZU1J2VP3OZMYAZC24EA2MNEPLN4XQHYBMJOB3EUY4GK2H&client_secret=FGTNJ055D2UTPZJIPYBPNPLWSNDSIJJU3WA4AENELLDX52ID&v=20180605&ll=43.6542599,-79.3606359&radius=500&limit=100'

In [80]:
results_api = requests.get(url).json()
results_api

{'meta': {'code': 200, 'requestId': '5d77abebf96b2c003984c21f'},
 'response': {'suggestedFilters': {'header': 'Tap to show:',
   'filters': [{'name': 'Open now', 'key': 'openNow'}]},
  'headerLocation': 'Corktown',
  'headerFullLocation': 'Corktown, Toronto',
  'headerLocationGranularity': 'neighborhood',
  'totalResults': 51,
  'suggestedBounds': {'ne': {'lat': 43.6587599045, 'lng': -79.3544279001486},
   'sw': {'lat': 43.6497598955, 'lng': -79.36684389985142}},
  'groups': [{'type': 'Recommended Places',
    'name': 'recommended',
    'items': [{'reasons': {'count': 0,
       'items': [{'summary': 'This spot is popular',
         'type': 'general',
         'reasonName': 'globalInteractionReason'}]},
      'venue': {'id': '54ea41ad498e9a11e9e13308',
       'name': 'Roselle Desserts',
       'location': {'address': '362 King St E',
        'crossStreet': 'Trinity St',
        'lat': 43.653446723052674,
        'lng': -79.3620167174383,
        'labeledLatLngs': [{'label': 'display',
 

In [82]:
def get_category_type(row):
    try:
        categories_list = row['categories']
    except:
        categories_list = row['venue.categories']
        
    if len(categories_list) == 0:
        return None
    else:
        return categories_list[0]['name']

In [84]:
venues = results_api['response']['groups'][0]['items']
    
nearby_venues = json_normalize(venues) # flatten JSON

# filter columns
filtered_columns = ['venue.name', 'venue.categories', 'venue.location.lat', 'venue.location.lng']
nearby_venues =nearby_venues.loc[:, filtered_columns]

# filter the category for each row
nearby_venues['venue.categories'] = nearby_venues.apply(get_category_type, axis=1)

# clean columns
nearby_venues.columns = [col.split(".")[-1] for col in nearby_venues.columns]

nearby_venues.head()

name            categories  \
0                                 Roselle Desserts                Bakery   
1                                    Tandem Coffee           Coffee Shop   
2  Toronto Cooper Koo Family Cherry St YMCA Centre  Gym / Fitness Center   
3                              Body Blitz Spa East                   Spa   
4                                   Impact Kitchen            Restaurant   

         lat        lng  
0  43.653447 -79.362017  
1  43.653559 -79.361809  
2  43.653191 -79.357947  
3  43.654735 -79.359874  
4  43.656369 -79.356980

In [85]:
print('{} venues were returned by Foursquare.'.format(nearby_venues.shape[0]))

51 venues were returned by Foursquare.


In [103]:
map_Toronto_nearby = folium.Map(location=[43.6523873, -79.3835641], zoom_start=13)

# add markers to map
for lat, lng,cat in zip(nearby_venues['lat'], nearby_venues['lng'],nearby_venues['categories']):
    label = folium.Popup(label, parse_html=True)
    folium.CircleMarker(
        [lat, lng],
        radius=5,
        popup=name,
        #color='blue',
        fill=True,
        fill_color='#3186cc',
        fill_opacity=0.7,
        parse_html=False,
        clustered_marker = True).add_to(map_Toronto_nearby)  
    
map_Toronto_nearby
